cmd/credentialUtil.go

```diff
@@ -257,35 +257,146 @@ type rawFromToInfo struct {
 	sourceSAS, destinationSAS string // Standalone SAS which might be provided
 }
 
-func getCredentialInfoForLocation(ctx context.Context, location common.Location, resource, resourceSAS string, isSource bool) (credInfo common.CredentialInfo, isPublic bool, err error) {
+const trustedSuffixesNameAAD = "trusted-aad-suffixes"
+const trustedSuffixesNameS3 = "trusted-s3-suffixes"
+const trustedSuffixesAAD = "*.core.windows.net;*.core.chinacloudapi.cn;*.core.cloudapi.de;*.core.usgovcloudapi.net"
+const trustedSuffixesS3 = "*.amazonaws.com;*.amazonaws.com.cn"
+
+// checkAuthSafeForTarget checks our "implicit" auth types (those that pick up creds from the environment
+// or a prior login) to make sure they are only being used in places where we know those auth types are safe.
+// This prevents, for example, us accidentally sending OAuth creds to some place they don't belong
+func checkAuthSafeForTarget(ct common.CredentialType, resource, extraSuffixesAAD, extraSuffixesS3 string) error {
+
+	getSuffixes := func(list string, extras string) []string {
+		extras = strings.Trim(extras, " ")
```

Comment:
>Trim(extras, " ") [](start = 19, length = 17)

strings.TrimSpace exists

Reply:
Oh!  I will remember that for next time. Thanks.